docs/diagrams/EditEventActivityDiagram.puml

```diff
@@ -0,0 +1,37 @@
+@startuml
+start
+:User wants to edit an event for a patient;
+:User runs the "edite" command with patient index, event index, new event name and new date / datetime of the event;
+if () then ([else])
+    :PatientSync throws an error;
+    :PatientSync adds error message to output;
+else ([Valid command format])
+    :PatientSync checks if the patient index is valid;
+    if () then ([else])
+        :PatientSync throws an error;
+        :PatientSync adds error message to output;
+    else ([Valid patient index])
+        :PatientSync checks if the event index is valid;
+        if () then ([else])
+            :PatientSync throws an error;
+            :PatientSync adds error message to output;
+        else ([Valid event index])
+            if () then ([else])
+                :PatientSync throws an error;
+                :PatientSync adds error message to output;
+            else ([Valid event details])
+                :PatientSync edits the specified event;
+                :PatientSync updates the set of events for the patient;
+                :PatientSync updates the displayed patient list;
+                if () then ([Duplicate event])
+                    :PatientSync adds duplicate event message to output;
+                else ([else])
```

Comment:
Regarding this, I understand that that is how the EditEventCommand works currently but it may be better to do this alter the code such that the duplicate event check is done before actually updating the event set. 

Although it does not really make a difference to the final logic, it would seem cleaner and make the flow more logical

Reply:
yes i agree!